redbeams/src/main/java/com/sequenceiq/redbeams/service/network/NetworkParameterAdder.java

```diff
@@ -74,4 +94,14 @@
         }
         return parameters;
     }
+
+    private String getAzureSubnetToUseWithPrivateEndpoint(DetailedEnvironmentResponse detailedEnvironmentResponse, DBStack dbStack) {
+        String subscriptionId = subnetListerService.getAzureSubscriptionId(detailedEnvironmentResponse.getCrn());
+        return subnetChooserService.chooseSubnetForPrivateEndpoint(
+                detailedEnvironmentResponse.getNetwork().getSubnetMetas().values(), dbStack, detailedEnvironmentResponse.getNetwork().isExistingNetwork())
+                .stream()
+                .findFirst()
+                .map(csn -> subnetListerService.expandAzureResourceId(csn, detailedEnvironmentResponse, subscriptionId))
+                .map(sn -> sn.getId()).orElse(null);
```

Comment:
if is handled if the return value is  "null"?

Reply:
So if no private endpoint is requested, then it is not used.
But if the subnet was changed after environment registration but before the postgres was started, then it gives an error. So a validation is needed for that, true.

Reply:
I rearranged it: value is only added if private endpoint is requested. And if in that case no network is found, then an exception is thrown. This is the case when between env registration and database creation the network policies have been changed on the network. Thanks for spotting it.